inject-remove-header/Readme.md

```diff
@@ -0,0 +1,491 @@
+In this case the backing Kafka is PLAINTEXT but the proxy is SASL_PLAIN.
+
+### Step 3: Start the environment
+
+Start the environment with
+
+```bash
+docker-compose up -d zookeeper kafka1 kafka2 conduktor-proxy kafka-client
+```
+
+### Step 4: Create topics
+
+We create topics using the Kafka console tools, the below creates a topic named `inject_header_topic`
+
+```bash
+docker-compose exec kafka-client \
+  kafka-topics \
+    --bootstrap-server conduktor-proxy:6969 \
+    --command-config /clientConfig/proxy.properties \
+    --create --if-not-exists \
+    --topic inject_header_topic
+```
+We create topics using the Kafka console tools, the below creates a topic named `remove_header_key_pattern_topic`
+
+```bash
+docker-compose exec kafka-client \
+  kafka-topics \
+    --bootstrap-server conduktor-proxy:6969 \
+    --command-config /clientConfig/proxy.properties \
+    --create --if-not-exists \
+    --topic remove_header_key_pattern_topic
+```
+We create topics using the Kafka console tools, the below creates a topic named `remove_header_value_pattern_topic`
+
+```bash
+docker-compose exec kafka-client \
+  kafka-topics \
+    --bootstrap-server conduktor-proxy:6969 \
+    --command-config /clientConfig/proxy.properties \
+    --create --if-not-exists \
+    --topic remove_header_value_pattern_topic
+```
+We create topics using the Kafka console tools, the below creates a topic named `remove_header_key_value_pattern_topic`
+
+```bash
+docker-compose exec kafka-client \
+  kafka-topics \
+    --bootstrap-server conduktor-proxy:6969 \
+    --command-config /clientConfig/proxy.properties \
+    --create --if-not-exists \
+    --topic remove_header_key_value_pattern_topic
+```
+List the created topics
+
+```bash
+docker-compose exec kafka-client \
+  kafka-topics \
+    --bootstrap-server conduktor-proxy:6969 \
+    --command-config /clientConfig/proxy.properties \
+    --list
+```
+
+### <a name="injectHeader"></a> Step 5: Inject Header
+
+The same REST API can be used to configure the inject header feature. 
+
+The command below will instruct Conduktor Proxy to inject headers with value user ip, tenant and proxy ip in records on topic `inject_header_topic`. 
+
+```bash
+docker-compose exec kafka-client curl \
+    --silent \
+    --request POST "conduktor-proxy:8888/tenant/1-1/feature/inject-header" \
+    --header 'Content-Type: application/json' \
+    --data-raw '{
+        "config": {
+            "topic": "inject_header_topic",
+            "keys": {
+              "X-RAW_KEY": "a value",
+              "X-USER_IP": "{{userIp}}",
+              "X-TENANT": "{{tenant}}",
+              "X-USER_IP_PROXY_IP_TENANT": "{{userIp}} to {{proxyIp}} of {{tenant}}"
+            }
+        },
+        "direction": "REQUEST",
+        "apiKeys": "PRODUCE"
+    }'
+```
+
+
+### Step 6: Produce data to the topic
+
+Let's produce a simple record to the `inject_header_topic` topic.
+
+```bash
+echo 'inject_header' | docker-compose exec kafka-client \
```

Comment:
`docker compose exec -T`
it needs to open a tty

Reply:
Thank you, I updated